sklearn/linear_model/logistic.py

```diff
@@ -1201,6 +1201,16 @@ def __init__(self, penalty='l2', dual=False, tol=1e-4, C=1.0,
         self.warm_start = warm_start
         self.n_jobs = n_jobs
 
+        if fit_intercept and intercept_scaling == 'warn':
```

Comment:
Please move all validation to `fit` rather than `__init__`.

Reply:
I had it initially in `fit`, but a test failed and complained that a public attribute is being changed in `fit`. Apparently the policy is that only private attributes can be changed and set in `fit` functions. What should I do about that test?